src/services/entity_builder.js

```diff
@@ -149,6 +149,21 @@ export default class EntityBuilder {
     return {...params};
   }
 
+  validateAndCastFindAssetsParams(params) {
+    const allowedParametersList = ['page', 'perPage', 'createdBy'];
+
+    validateFieldsConstrainedToSet(params, allowedParametersList);
+
+    params.page = validateIntegerParameterAndCast(params.page, 'page');
+    params.perPage = validateIntegerParameterAndCast(params.perPage, 'perPage');
+    if (params.createdBy) {
+      validateIsAddress(params.createdBy);
+    }
+
+    return {...params};
+  }
+
```

Comment:
We already got method 'validateAndCastFindEventsParams' that is really, really similar.
```
  validateAndCastFindEventsParams(params) {
    const allowedParametersList = ['assetId', 'fromTimestamp', 'toTimestamp', 'page', 'perPage', 'createdBy', 'data'];

    validateFieldsConstrainedToSet(params, allowedParametersList);

    this.ensureGeoLocationParamsCorrectlyPlaced(params);

    params.fromTimestamp = validateIntegerParameterAndCast(params.fromTimestamp, 'fromTimestamp');
    params.toTimestamp = validateIntegerParameterAndCast(params.toTimestamp, 'toTimestamp');
    params.page = validateIntegerParameterAndCast(params.page, 'page');
    params.perPage = validateIntegerParameterAndCast(params.perPage, 'perPage');
    if (params.createdBy) {
      validateIsAddress(params.createdBy);
    }

    return {...params};
  }
```
We also have something similar for account querying.
```
  validateAndCastFindAccountParams(params) {
    const allowedParametersList = ['accessLevel', 'page', 'perPage', 'registeredBy'];

    validateFieldsConstrainedToSet(params, allowedParametersList);

    params.accessLevel = validateIntegerParameterAndCast(params.accessLevel, 'accessLevel');
    params.page = validateIntegerParameterAndCast(params.page, 'page');
    params.perPage = validateIntegerParameterAndCast(params.perPage, 'perPage');
    if (params.registeredBy) {
      validateIsAddress(params.registeredBy);
    }

    return {...params};
  }
```
Maybe we should think about moving some stuff from 'EntityBuilder' and 'AccountAccessDefinitions' to some kind of query utils, or query validators?

Reply:
Shall we add a sperate ticket for that?
Sth like:
```js
validateAndCastFindEventsParams(params) {
  validate(params, {
    page: integer, 
    perPage: integer, 
    createdBy: {type: address, required: false}
  });
}
```